protected/humhub/exceptions/InvalidArgumentException.php

```diff
@@ -0,0 +1,19 @@
+<?php
+
+/*
+ * @link      https://www.humhub.org/
+ * @copyright Copyright (c) 2023 HumHub GmbH & Co. KG
+ * @license   https://www.humhub.com/licences
+ */
+
+namespace humhub\exceptions;
+
+use yii\base\InvalidArgumentException as BaseInvalidArgumentException;
+
+/**
+ * @since 1.15
+ */
+class InvalidArgumentException extends BaseInvalidArgumentException
```

Comment:
@martin-rueegg I'm not sure, if I'm a big fan of the new exception classes. 

e.g. so far in the IDE class picker, I simply searched for a suitable exception class type and then `throw new InvalidArgumentException($message = 'Invalid user given!')`. When this was thrown, you see a stack trace.

Now there are own Humhub "InvalidArgumentException" classes, instead of the well known`InvalidArgumentException(message)` style,  with new parameters `InvalidArgumentException(__METHOD__, parameter, valid, given, ...)`. For me and other devs, now a documentation is probably required.

I understand there are some benefits here, but for me it's also a bit over engineered.

To sum up, big cons for me:

- Complexity (Parameters & co). 
- Different Constructor Signature between `yii\base\InvalidArgumentException`  and `humhub\exceptions\InvalidArgumentException` 
- When accepted, we should use this more...

Reply:
Thank you for your review, @luke-, and your comments.

I appreciate it takes some time to craft the new `InvalidArgumentException(__METHOD__, parameter, valid, given, ...)`.

However, this is a one-time effort and once you're used to the parameters (which appart from `$parameter` and `$valid` seem to be quite self-explanatory to me), it's not costing much more time than writing `InvalidArgumentException('Invalid user given!')`. And for me the benefit are 
- When the error is thrown, I see what value was received and why it was the wrong $value or value type
- Without this, I have to dig through the code of the called function/method to understand which values/types are allowed (doc comments unfortuanately are not always up-to-date and are only visible in the source code).
- The source of the value is often many calls earlier. There can be easily 10-20 function calls in between and the stack trace usually shows the defining class, not the class of the respective instance, and hence I find it really time-consuming to track down the source code and where it came from. In my experience, giving a more verbose output in the error message helps with this.
- Also, the "invalid user" example may be too simple, and for such a use-case, it may be more suitable to use the `yii\base\InvalidArgumentException`. However, if it is a `state` (see new status interface and implementation) then it makes sense to see what states are available. In this case, this exception lists the received value along with the array of allowed values (which is automatically retrieved from the defining `StateService`). Or if one is validating a `$config` parameter, providing the list of allowed parameters in the Exception message can e.g. show a typo immediately, without me going to search the source code
- So the main advantage for me is, that I have a little more effort to write the Exception message, but often save time when debugging the cause. Which, as opposed to the former, the latter happens many times for a given method.
- `InvalidArgumentException` is extended by `InvalidArgumentTypeException`, which has previously been accepted.
- the `$method` parameter could be left away and be replaced by a trace inspection. However, sometimes it's helpful to pass the name of the method that has the restriction to a sub-method that actually does (part of) the verification. But again, it's a small habit to just type `_`, `_`, `M`, `tab`, `,` and go on ... (PhpStorm also provides facilites to store snippets. It may be a useful snippet for a shared PhpStorm-Config-Repository.)
- Happy to extend some documentation in the Exception class.

Reply:
I understand that such exception are really handy too. 

Here is what I would prefer:

- Rename the `InvalidArgumentException` to `InvalidArgumentValueException`. To reflect it's difference between `yii\base\InvalidArgumentException`.

- Can we simplify them? For me, this is way too unintuitive: `throw new InvalidArgumentException(__METHOD__, [2 => '$type'], ['string', 'string[]'], $type, false, 1);`

What about just:
	- `InvalidArgumentValueException(string $argumentName, ?$validValues, $?givenValue);`
	- `InvalidArgumentTypeException($argumentName, ?$validTypes, ?$givenType)`
	
e.g. `throw new InvalidArgumentValueException('type', ['string', 'string[]'], $type);`

- Remove Traits when possible

Reply:
@luke- I've moved the amendment of the Exceptions to
- #6566
And may have found a good compromise in regards to compatibility with `yii\base\InvalidArgumentException`.